tcases-lib/src/main/java/org/cornutum/tcases/VarBinding.java

```diff
@@ -41,22 +41,42 @@ public VarBinding( String varName, String valueName)
    */
   public VarBinding( String varName, String varType, String valueName)
     {
-    setVar( varName);
-    setType( varType);
-    setValue( valueName);
-    setValueValid( true);
+    this( varName, varType, valueName, false);
     }
   
   /**
    * Creates a new VarBinding object.
    */
   public VarBinding( IVarDef varDef, VarValueDef valueDef)
     {
-    this( varDef.getPathName(), varDef.getType(), valueDef.getName());
+    this( varDef.getPathName(), varDef.getType(), valueDef.getName(), valueDef.isNA());
     setValueValid( valueDef.getType().isValid());
     setVarDef( varDef);
     }
   
+  /**
+   * Creates a new "not applicable" VarBinding object.
+   */
+  public static VarBinding notApplicable( String varName, String varType)
+    {
+    return new VarBinding( varName, varType, null, true);
+    }
+  
+  /**
+   * Creates a new VarBinding object.
+   */
+  private VarBinding( String varName, String varType, String valueName, boolean valueNA)
```

Comment:
For style might prefer inverse boolean "isApplicable" as last argument, to avoid double negation logic (false == not not applicable).

Reply:
Actually this could apply to many methods / variables. E. isApplicable() instead of isValueNA()

Reply:
Just brainstorming: Maybe polymorphism could be used with new classes VarNADef and NAVarBinding, instead of having value classes with booleans and code which has to manage if/else conditions in many places to avoid inconsistent instance state.

Reply:
> Maybe polymorphism could be used with new classes VarNADef, etc.

That's an interesting idea. Let me see how that might work out.